mlflow/deployments/databricks/__init__.py

```diff
@@ -1,39 +1,135 @@
+from mlflow.utils.rest_utils import augmented_raise_for_status, http_request
 
 
 class DatabricksDeploymentClient(BaseDeploymentClient):
     def create_deployment(self, name, model_uri, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def update_deployment(self, name, model_uri=None, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def delete_deployment(self, name, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def list_deployments(self, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def get_deployment(self, name, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
-    def predict(self, deployment_name=None, inputs=None, endpoint=None):
-        raise NotImplementedError("TODO")
+    def _call_endpoint(
+        self,
+        *,
+        method: str,
+        prefix: str = "/api/2.0",
+        route: Optional[str] = None,
+        json_body: Optional[Dict[str, Any]] = None,
+        timeout: int = MLFLOW_HTTP_REQUEST_TIMEOUT.get(),
+    ):
+        call_kwargs = {}
+        if method.lower() == "get":
+            call_kwargs["params"] = json_body
+        else:
+            call_kwargs["json"] = json_body
+
+        response = http_request(
+            host_creds=get_databricks_host_creds(self.target_uri),
+            endpoint=posixpath.join(prefix, "serving-endpoints", route or ""),
+            method=method,
+            timeout=timeout,
+            raise_on_status=False,
+            retry_codes=MLFLOW_DEPLOYMENT_CLIENT_REQUEST_RETRY_CODES,
+            **call_kwargs,
+        )
+        augmented_raise_for_status(response)
+        return response.json()
 
+    def predict(self, deployment_name=None, inputs=None, endpoint=None):
+        """
+        TODO
+        """
+        return self._call_endpoint(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body=inputs,
+            timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT,
+        )
+
+    # TODO: Return a DatabricksEndpoint object that wraps the response dictionary and exposes some
+    # properties nicely like name and endpoint type (https://github.com/mlflow/mlflow/pull/10425/files#r1396514655)
     def create_endpoint(self, name, config=None):
```

Comment:
@dbczumar We could a return dict-like object here:

```
class AttrDict(dict):
    """
    Examples
    --------
    >>> d = AttrDict({'a': 1, 'b': 2})
    >>> d.a
    1
    >>> d = AttrDict({'a': 1, 'b': {'c': 3, 'd': 4}})
    >>> d.b.c
    3
    """

    def __getattr__(self, attr):
        value = self[attr]
        if isinstance(value, dict):
            return AttrDict(value)
        return value
```

Reply:
Agreed from Corey!